launcher/minecraft/MinecraftInstance.cpp

```diff
@@ -1014,8 +1014,18 @@ QMap<QString, QString> MinecraftInstance::createCensorFilterFromSession(AuthSess
 
 MessageLevel::Enum MinecraftInstance::guessLevel(const QString& line, MessageLevel::Enum level)
 {
-    QRegularExpression re("\\[(?<timestamp>[0-9:]+)\\] \\[[^/]+/(?<level>[^\\]]+)\\]");
-    auto match = re.match(line);
+    if (line.contains("overwriting existing"))
+        return MessageLevel::Fatal;
+
+    // NOTE: this diverges from the real regexp. no unicode, the first section is + instead of *
+    static const QRegularExpression JAVA_EXCEPTION(
+        R"(Exception in thread|...\d more$|(\s+at |Caused by: )([a-zA-Z_$][a-zA-Z\\d_$]*\.)+[a-zA-Z_$][a-zA-Z\\d_$]*)");
+
+    if (line.contains(JAVA_EXCEPTION))
+        return MessageLevel::Error;
```

Comment:
Shouldn't this check be last? some errors are emitted as warnings and this will override the level.

Reply:
It does look like I missed out this pattern:

```
([a-zA-Z_$][a-zA-Z\\d_$]*\\.)+[a-zA-Z_$]?[a-zA-Z\\d_$]*(Exception|Error|Throwable)
```
but it's quite ugly to duplicate the regex for a java symbol lol